packages/data/addon/serializers/bard-metadata.js

```diff
@@ -14,13 +14,15 @@ export default EmberObject.extend({
    * @param rawTables {Array} - array of table objects
    * @returns {Object} - normalized table object
    */
-  _normalizeTable: function(rawTables) {
+  _normalizeTable: function(rawTables, source) {
     // build dimension and metric arrays
     let dimensions = [],
       metrics = [],
       tables = rawTables.map(table => {
         let timeGrains = table.timeGrains.map(timegrain => {
-          dimensions = dimensions.concat(timegrain.dimensions);
+          dimensions = dimensions.concat(
+            timegrain.dimensions.map(dimension => Object.assign({}, dimension, { source }))
```

Comment:
Can we namespace the metadata object ids witha datasource prefiex such as:
```js
`${source}.${tableName}`
```

Reply:
I put this functionality in the keg.